sds/src/secondary_validation/romanian_personal_numeric_code.rs

```diff
@@ -0,0 +1,79 @@
+use crate::secondary_validation::Validator;
+pub struct RomanianPersonalNumericCode;
+
+const CHECKSUM_WEIGHTS: &[u32] = &[2, 7, 9, 1, 4, 6, 3, 5, 8, 2, 7, 9];
+
+impl Validator for RomanianPersonalNumericCode {
+    fn is_valid_match(&self, regex_match: &str) -> bool {
+        if !regex_match.is_ascii() {
+            return false;
+        }
+        for c in regex_match.chars() {
+            if !c.is_ascii_digit() {
+                return false;
+            }
+        }
+
+        let mut calculated_checksum = 0;
+        for (i, c) in regex_match.chars().take(12).enumerate() {
+            let digit = if let Some(digit) = c.to_digit(10) {
+                digit
+            } else {
+                return false;
+            };
+            calculated_checksum += CHECKSUM_WEIGHTS[i] * digit;
+        }
+
+        calculated_checksum %= 11;
+        if calculated_checksum == 10 {
+            calculated_checksum = 1;
+        }
+
+        let actual_checksum = if let Some(c) = regex_match.chars().last().unwrap().to_digit(10) {
+            c
+        } else {
+            return false;
+        };
+
+        calculated_checksum == actual_checksum
```

Comment:
We should ideally avoid unwrap. 
Assuming the digits have been extracted from the input, this could be simplified with a `next`:

```rust
        if let Some(actual_checksum) = digits.next() {
            return calculated_checksum == actual_checksum;
        } 
        false
```